Document getPaymentConfiguration (#24)
MewsSystems/gitbook-distributor-guide: distributor-api-v1/operations.md

A localized text is an object of the property values localized into languages supported by hotel, indexed by language codes.

#### PaymentGateway   <a id="payment-gateway"></a>

If the hotel does not use any payment gateway, the value is null. If it does, then you should use a specific api call and the gateway’s library to encode credit card data. The main purpose of a payment gateway is to securely obtain credit card of the customer before a reservation is created. You can decide not to support any of them and just ignore it, in which case reservations are created with note about missing credit card.

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `PaymentCardStorageType` | string [PaymentCardStorageType](operations.md#payment-card-storage-type) | required | Type of the payment card storage used by enterprise. |

Author: yep

| `IsMerchant` | boolean | required | Whether the gateway is processed through Mews Merchant or not. |

> | `SupportedCreditCardTypes` | string [CreditCardType](operations.md#credit-card-type) | required | The list of supported payment cards, should be used to enhance UX. |

Reviewer: How can property of type string contain "list of supported..."?

Author: It is string that contains enum value. Same as here:
https://app.gitbook.com/@mews-systems/s/connector-api/operations/customers#gender

Reviewer: But the description cannot be "The **list** of supported payment cards."

> | `PublicKey` | boolean | required | Merchant identifier for which PCI proxy Iframe is connected. |

Author: ```suggestion
| `PublicKey` | string | required | Merchant identifier for which PCI proxy Iframe is connected. |
```

> | `DefaultCurrencyCode` | string | required | Currency of default payment gateway. |

Author: ```suggestion
| `DefaultCurrencyCode` | string | required | Currency code of default payment gateway in ISO format. |
```

> | `DefaultCurrencyCode` | string | required | Currency of default payment gateway. |

Author: choose which one sounds better and commit it :D . Both are equally good but i cant decide :D

#### PaymentCardStorageType

* Adyen

> * Stripe

Author: yep, tehy are PaymentCardStorageType

A localized text is an object of the property values localized into languages supported by hotel, indexed by language codes.

> #### PaymentGateway   <a id="payment-gateway"></a>

Reviewer 2: But now it's still inconsistent, for example Product charging, Product posting. All these should be camelCase one word. What do you think? @dobras  @dmarchuk

Reviewer 3: @dobras @zdrazil Most of the things in this document are one word PascalCase, so I think we should make it all consistent in that way. Also in the API everything is PascalCase, so I think it makes sense to keep it that way.

Reviewer 2: Sorry, yeah I've meant PascalCase. But Product charging, Product posting. Those are not written in PascalCase. So it should be unified. What do you think?

Author: Added to the followup issue

| :--- | :--- | :--- | :--- |
| `OccupancyPrices` | array of [RoomOccupancyAvailability](operations.md#roomoccupancyavailability) | required | Pricing information. |

## Get Payment Configuration   <a id="get-payment-configuration"></a>

> ### Request `[PlatformAddress]/api/distributor/v1/hotels/getPaymentConfiguration`   <a id="request-getpaymentconfiguration"></a>

Reviewer 3: @dobras 📝 Please create a follow up task where we should fix this formatting for each `Request` (adding a space).

Author: https://mews.myjetbrains.com/youtrack/issue/TDD-1726

| :--- | :--- | :--- | :--- |
| `OccupancyPrices` | array of [RoomOccupancyAvailability](operations.md#roomoccupancyavailability) | required | Pricing information. |

## Get Payment Configuration   <a id="get-payment-configuration"></a>

> ### Request`[PlatformAddress]/api/distributor/v1/hotels/getPaymentConfiguration`   <a id="request-getpaymentconfiguration"></a>

Reviewer 3: @dobras Maybe try to put the URL of request to a new line so the URL is not broken to the next line in default browser size? Not sure if it will be better, but I don't like the URL spreading to two lines on a normal window size.

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `HotelId` | string | required | Unique identifier of hotel. |
| `Client` | string | required | Identification of the client as described in [Authorization](https://mews-systems.gitbook.io/distributor-guide/distributor-api-v1/authorization). |
| `Session` | number | required | Session number. |
| `LanguageCode` | string | required | Language code. |
| `CultureCode` | string | required | Culture Code. |

### Response  <a id="response-getpaymentconfiguration"></a>

```javascript
{
    "PaymentGateway": {
        "PaymentGatewayType": "PciProxy",
        "PaymentCardStorageType": "PciProxy",
        "IsMerchant": true,
        "SupportedCreditCardTypes": [
            "MasterCard",
            "Visa"
        ],
        "PublicKey": "1100016614",
        "DefaultCurrencyCode": "EUR"
    },
    "SurchargeConfiguration": {
        "SurchargeServiceId": null,

>         "SurchargeFees": {}

Reviewer 3: @dobras ⛏ Here we should probably put values from a hotel that has this defined.

    }
}
```

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `PaymentGateway` | [PaymentGateway](operations.md#payment-gateway) | required | Object that describes Payment gateway of the enterprise. |

Reviewer 3: @dobras Why capital `P` here? It should either be lowercase or `PaymentGateway`.

| `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object that describes surcharge configuration that enterprise uses. |

> #### Surcharge Configuration <a id="surcharge-configuration"></a>

Reviewer 3: @dobras There shouldn't be any space here, just `SurchargeConfiguration`. It should be consistent with how it's called in `Property` column in the table above. This is also the case for `PaymentGateway`, which could be fixed in this PR as well - that would make it consistent across this document.

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `SurchargeServiceId` | string | optional | Unique identifier of surcharge service, if enterprise uses surcharge feature this Id will be provided. |

> | `SurchargeFees` | dictionary of string/decimal | optional | Dictionary of surcharge fees where string is `PatmentCardType`(such as Visa, MasterCard, Amex ...) and decimal is amount of the surcharge fee itself. |

Reviewer 2: I don't think we describe it as dictionary instead it should be maybe kind of similar to for example 
![image](https://user-images.githubusercontent.com/6526378/74835315-54cb7b80-531d-11ea-8644-a383c4fc0689.png)

link with a special section where this is described.

And also it's better to call in object, since that's what FE has? LocalizedText is described like that.

> | `SurchargeFees` | dictionary of string/decimal | optional | Dictionary of surcharge fees where string is `PatmentCardType`(such as Visa, MasterCard, Amex ...) and decimal is amount of the surcharge fee itself. |

Reviewer 3: @dobras 
- `decimal` does not exist in JSON, just `number`.
- ⛏I'm not sure about using `dictionary` in here, maybe `collection of key-value pairs` would be better?

> | `SurchargeFees` | dictionary of string/decimal | optional | Dictionary of surcharge fees where string is `PatmentCardType`(such as Visa, MasterCard, Amex ...) and decimal is amount of the surcharge fee itself. |

Reviewer 3: @dobras Here we should actually have possible keys of the `SurchargeFees` object, so API users would know what can they expect in this object.

        "PaymentGatewayType": "PciProxy",
        "PaymentCardStorageType": "PciProxy",
        "IsMerchant": true,
        "SupportedCreditCardTypes": [
            "MasterCard",
            "Visa"
        ],
        "PublicKey": "1100016614",
        "DefaultCurrencyCode": "EUR"
    },
    "SurchargeConfiguration": {
        "SurchargeServiceId": "74d5eb0a-784a-4870-b34a-ab6500a1136e",
        "SurchargeFees": {
            "MasterCard": 0.02,
            "Visa": 0.01,
            "Amex": 0.0125
        }
    }
}
```

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `PaymentGateway` | [PaymentGateway](operations.md#payment-gateway) | required | Object that describes payment gateway of the enterprise. |
| `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object that describes surcharge configuration that enterprise uses. |

#### SurchargeConfiguration <a id="surcharge-configuration"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `SurchargeServiceId` | string | optional | Unique identifier of surcharge service, if enterprise uses surcharge feature this Id will be provided. |

> | `SurchargeFees` | collection of key-value pairs where key is string and value is number | optional | Collection of key-value pairs where key is string representation of CreditCardType (such as Visa, MasterCard, Amex ...) and value is amount of the surcharge fee itself. |

Reviewer 3: @dobras I think we should also add some information into description that this will be an empty object if there is no surcharge service configured.

| `HotelId` | string | required | Unique identifier of hotel. |
| `Session` | number | required | Session number. |
| `LanguageCode` | string | required | Language code. |
| `CultureCode` | string | required | Culture Code. |

### Response  <a id="response-getpaymentconfiguration"></a>

```javascript
{
    "PaymentGateway": {
        "PaymentGatewayType": "PciProxy",
        "PaymentCardStorageType": "PciProxy",
        "IsMerchant": true,
        "SupportedCreditCardTypes": [
            "MasterCard",
            "Visa"
        ],
        "PublicKey": "1100016614",
        "DefaultCurrencyCode": "EUR"
    },
    "SurchargeConfiguration": {
        "SurchargeServiceId": "74d5eb0a-784a-4870-b34a-ab6500a1136e",
        "SurchargeFees": {
            "MasterCard": 0.02,
            "Visa": 0.01,
            "Amex": 0.0125
        }
    }
}
```

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `PaymentGateway` | [PaymentGateway](operations.md#payment-gateway) | required | Object that describes payment gateway of the enterprise. |

> | `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object that describes surcharge configuration that enterprise uses. |

Reviewer 3: ```suggestion
| `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object describing surcharge configuration used by the enterprise. |
```

#### SurchargeConfiguration <a id="surcharge-configuration"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `SurchargeServiceId` | string | optional | Unique identifier of surcharge service, if enterprise does not use surcharge feature this Id will be `null`. |

> | `SurchargeFees` | collection of key-value pairs where key is string and value is number | optional | Collection of key-value pairs where key is string representation of [CreditCardType](operations.md#creditcardtype) and value is amount of the surcharge fee itself. If enterprise does not use surcharge feature this will be an empty object. |

Reviewer 3: @dobras What do you think about the change from `dictionary` to `key-value pairs`? I thought about it a bit more and maybe it would actually be better if we just write `object` as @zdrazil suggested, so here it would be e.g. `object where each key is string and value is number`. What do you think? Or do you have any better suggestions? Using `collection` does sound a bit weird in JSON.

Author: Im not an expert on json but this just looks like an object that has properties where name of the property is a string and value is number. And it has a lot of them. It does not have `][` these brackets to indicate collection.

> | `SurchargeFees` | collection of key-value pairs where key is string and value is number | optional | Collection of key-value pairs where key is string representation of [Creditcard type](operations.md#creditcard-type) and value is amount of the surcharge fee itself. If enterprise does not use surcharge feature this will be an empty object. |

Reviewer 3: @dobras Everywhere we write these things as pascal case (`CreditCard`) so I think here it should be named like that as well.

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `Client` | string | required | Identification of the client as described in [Authorization](https://mews-systems.gitbook.io/distributor-guide/distributor-api-v1/authorization). |
| `HotelId` | string | required | Unique identifier of hotel. |

> | `Session` | number | required | Session number. |

Reviewer 2: why are the dots added if it's not a sentence?

Reviewer 2: After we decide, please create a follow up task to fix the dots everywhere to make it consistent.

Author: Will create an issue for it.

| `LanguageCode` | string | required | Language code. |
| `CultureCode` | string | required | Culture Code. |

### Response  <a id="response-getpaymentconfiguration"></a>

```javascript
{
    "PaymentGateway": {
        "PaymentGatewayType": "PciProxy",
        "PaymentCardStorageType": "PciProxy",
        "IsMerchant": true,

>         "SupportedCreditCardTypes": [

Reviewer 2: maybe describe it like 
```
[CreditCardType](operations.md#creditcardtype)
```

            "MasterCard",
            "Visa"
        ],

>         "PublicKey": "1100016614",

Reviewer 2: section:
```
#### Payment Gateway   <a id="payment-gateway"></a>
```
doesn't describe what's described here in the response, things such as `PublicKey`, `PaymentCardStorageType` etc.

Author: Will extend PaymentGateway section.

>         "PublicKey": "1100016614",

Reviewer 2: The public key is fake, we'll not leak something important, right?

Author: Fake

        "DefaultCurrencyCode": "EUR"
    },
    "SurchargeConfiguration": {
        "SurchargeServiceId": "74d5eb0a-784a-4870-b34a-ab6500a1136e",
        "SurchargeFees": {
            "MasterCard": 0.02,
            "Visa": 0.01,
            "Amex": 0.0125
        }
    }
}
```

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `PaymentGateway` | [PaymentGateway](operations.md#payment-gateway) | required | Object that describes payment gateway of the enterprise. |
| `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object describing surcharge configuration used by the enterprise. |

#### SurchargeConfiguration <a id="surcharge-configuration"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `SurchargeServiceId` | string | optional | Unique identifier of surcharge service, if enterprise does not use surcharge feature this Id will be `null`. |

Reviewer 2: what if we come up with a solution how to write the optional value not in the description, but under type. Or is optional always null?

Author: Optional can be null but it does not have to be.

Author: When there is object coming from the API even if it is empty, it will be required

> | `SurchargeFees` | object containing key-value pairs where key is string and value is number | optional | Object containing key-value pairs where key is string representation of [CreditCardType](operations.md#creditcardtype) and value is amount of the surcharge fee itself. If enterprise does not use surcharge feature this will be an empty object. |

Reviewer 2: What about making this a separate section 
```
| `SurchargeFees` | [SurchargeFees](operations.md#surcharge-fees) | optional |....
```
?

Author: I tried the same with Amounts and did not work out

Author: Solved in person

Author: Solved in person

> | `SurchargeFees` | object containing key-value pairs where key is string and value is number | optional | Object containing key-value pairs where key is string representation of [CreditCardType](operations.md#creditcardtype) and value is amount of the surcharge fee itself. If enterprise does not use surcharge feature this will be an empty object. |

Reviewer 2: if it can be empty object, it's not optional, right?

#### CreditCardType 

> * MasterCard

Reviewer 2: What's the point of making the list if we don't describe everything? What could this people use it for?

Author: Discussed in person

```javascript
{
    "PaymentGateway": {
        "PaymentGatewayType": "PciProxy",
        "PaymentCardStorageType": "PciProxy",
        "IsMerchant": true,
        "SupportedCreditCardTypes": [
            "MasterCard",
            "Visa"
        ],
        "PublicKey": "1100116614",
        "DefaultCurrencyCode": "EUR"
    },
    "SurchargeConfiguration": {
        "SurchargeServiceId": "74d5eb0a-784a-4870-b34a-ab6500a1136e",
        "SurchargeFees": {
            "MasterCard": 0.02,
            "Visa": 0.01,
            "Amex": 0.0125
        }
    }
}
```

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `PaymentGateway` | [PaymentGateway](operations.md#payment-gateway) | required | Object that describes payment gateway of the enterprise. |
| `SurchargeConfiguration` | [SurchargeConfiguration](operations.md#surcharge-configuration) | required | Object describing surcharge configuration used by the enterprise. |

#### SurchargeConfiguration <a id="surcharge-configuration"></a>

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |
| `SurchargeServiceId` | string | optional | Unique identifier of surcharge service. |

> | `SurchargeFees` | [SurchargeFees](operations.md#surcharge-fees) | required | Surcharge fees are additional fees charged by credit card company. |

Reviewer 2: ```suggestion
| `SurchargeFees` | [SurchargeFees](operations.md#surcharge-fees) | required | Surcharge fees are additional fees charged by payment card company. |
```

#### SurchargeFees

|  | Property | Type | Description |
| :--- | :--- | :--- | :--- |

> | `Key` | string [CreditCardType](operations.md#credit-card-type) | required | Credit card types. |

Reviewer: This is singular - Credit Card type.
